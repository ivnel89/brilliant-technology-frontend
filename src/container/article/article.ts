import Api from "../../api";
import * as $ from "jquery";
import { Article } from "../../api/contract";

const api = new Api();

export const getArticleId = ()=> {
    const searchParams = new URLSearchParams(window.location.search)
    return searchParams.get("id")
} 

export const renderArticle = async () => {
    const articleId = getArticleId();
    const article: Article = await api.fetchArticleById(articleId);
    const articleDatePretty = new Date(article.createdDate).toLocaleString()
    const articleMarkup: JQuery<HTMLElement> = $(`
    <article>
        <div>
            Written by ${article.author.firstName} ${article.author.lastName}
            <br/>
            Published on ${articleDatePretty}
        </div>
        <p>
        ${article?.content}
        </p>
    </article>
    `)
    $(`#root`).append(articleMarkup);

}

renderArticle();