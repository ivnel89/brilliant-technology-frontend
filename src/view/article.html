<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8"/>
    <title>My Article Website</title>
    <link rel="stylesheet" href="../styles.css" />
    <script type="module" src="../container/article/article.ts"></script>
  </head>
  <body class="container mx-auto px-4" >
    <div id="root"></div>
  </body>
</html>