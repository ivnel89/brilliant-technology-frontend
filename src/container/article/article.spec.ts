import { renderArticle } from './article';
import * as $ from "jquery";

jest.mock('../../api', () => {
    return {
        default: jest.fn().mockImplementation(() => {
            return {
                fetchArticleById: jest.fn().mockResolvedValue(
                {
                  author: {
                    displayPicture: "www.example.com/img.png",
                    firstName: "Ben",
                    id: "mockAuthorId",
                    lastName: "Cook",
                  },
                  comments: [],
                  content: "Title",
                  createdDate: new Date(),
                  id: "mockArticleId",
                },
              ),
            };   
        })
    }
  });
describe(`renderArticle`, () => {
    it(`should render the article`, async () => {
        document.body.innerHTML = `
        <div id="root"></div>
      `;
        await renderArticle();
        expect($("#root").html()).toMatchSnapshot()
    } )
})