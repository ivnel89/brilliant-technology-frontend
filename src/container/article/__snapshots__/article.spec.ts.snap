// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderArticle should render the article 1`] = `
"<article>
        <div>
            Written by Ben Cook
            <br>
            Published on 24/04/2022, 00:20:24
        </div>
        <p>
        Title
        </p>
    </article>"
`;
